agent/checks/check.go

```diff
@@ -321,6 +325,23 @@ type CheckHTTP struct {
 	stop       bool
 	stopCh     chan struct{}
 	stopLock   sync.Mutex
+
+	// Set if checks are exposed through Connect proxies
+	// If set, this is the target of check()
+	ProxyHTTP string
+}
+
+func (c *CheckHTTP) CheckType() structs.CheckType {
+	return structs.CheckType{
+		CheckID:       c.CheckID,
+		HTTP:          c.HTTP,
+		Method:        c.Method,
+		Header:        c.Header,
+		Interval:      c.Interval,
+		ProxyHTTP:     c.ProxyHTTP,
+		Timeout:       c.Timeout,
+		OutputMaxSize: c.OutputMaxSize,
+	}
```

Comment:
🤔 wonder if it's worth commenting in `structs.CheckType` that this exists in case new fields are added there in the future?

Reply:
These conversions hard to maintain - 👍 for everything we can do to help the next person.

Reply:
Just added a note